sentry-android-core/src/main/java/io/sentry/android/core/AndroidOptionsInitializer.java

```diff
@@ -48,9 +48,15 @@ static void init(
       @NotNull Context context,
       final @NotNull ILogger logger) {
     Objects.requireNonNull(context, "The context is required.");
-    context =
-        Objects.requireNonNull(
-            context.getApplicationContext(), "The application context is required.");
+
+    if (!(context instanceof Application)) {
```

Comment:
Maybe this is not needed? Wouldn't it be easier to just say "if the application context is not null I will use that, otherwise I will stick with what you gave me"? Don't get me wrong, this is ok. Just wondering about maintainability.

Reply:
@ninniuz yeah, I thought to use the given Context, and only overwrite if it's actually needed, even though if the context is already an Application, `getApplicationContext` returns the same `Application` instance right.

Reply:
Yes, that's why I think it is safe for both SDK users and the SDK itself to use `context.getApplicationContext()` if it is not `null` and fallback to the provided one otherwise. Of course you would have to keep the other checks for integrations as you already pointed out.

Reply:
yep, done, thanks :)